internal/util/helpers_test.go

```diff
@@ -4,18 +4,765 @@
+			expectedError: "error unmarshaling credentials: invalid character 'i'",
+		},
+		{
+			name:          "Error reading credentials file",
+			fileContent:   "",
+			expectedError: "error reading credentials file",
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			testDir := t.TempDir()
+			SetEnvVar(t, "XDG_CONFIG_HOME", testDir)
+			// Create the minder directory inside the temp directory
+			minderDir := filepath.Join(testDir, "minder")
+			err := os.MkdirAll(minderDir, 0750)
+			if err != nil {
+				t.Fatalf("failed to create minder directory: %v", err)
+			}
+
+			filePath := filepath.Join(minderDir, "credentials.json")
+
+			if tt.fileContent != "" {
+				// Create a temporary file with the specified content
+				err := os.WriteFile(filePath, []byte(tt.fileContent), 0600)
+				if err != nil {
+					t.Fatalf("failed to write test file: %v", err)
+				}
+				// Print the file path and content for debugging
+				t.Logf("Test %s: written file path %s with content: %s", tt.name, filePath, tt.fileContent)
+			} else {
+				// Print the file path for debugging
+				t.Logf("Test %s: file path %s not created as file content is empty", tt.name, filePath)
+			}
+
+			result, err := util.LoadCredentials()
+			if tt.expectedError != "" {
+				if err == nil || !strings.HasPrefix(err.Error(), tt.expectedError) {
+					t.Errorf("expected error matching %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken || !result.AccessTokenExpiresAt.Equal(tt.expectedResult.AccessTokenExpiresAt) {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+
+		})
+
+	}
+}
+
+type TestCase struct {
+	name         string
+	token        string
+	issuerUrl    string
+	clientId     string
+	tokenHint    string
+	expectedPath string
+	expectError  bool
+	createServer func(t *testing.T, tt TestCase) *httptest.Server
+}
+
+func createTestServer(t *testing.T, tt TestCase) *httptest.Server {
+	t.Helper()
+	return httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+		err := r.ParseForm()
+		require.NoError(t, err, "error parsing form")
+
+		require.Equal(t, tt.clientId, r.Form.Get("client_id"))
+		require.Equal(t, tt.token, r.Form.Get("token"))
+		require.Equal(t, tt.tokenHint, r.Form.Get("token_type_hint"))
+	}))
+}
+
+// TestRevokeToken tests the RevokeToken function
+func TestRevokeToken(t *testing.T) {
+	t.Parallel()
+	tests := []TestCase{
+		{
+			name:         "Valid token revocation",
+			token:        "test-token",
+			issuerUrl:    "http://localhost:8081",
+			clientId:     "minder-cli",
+			tokenHint:    "refresh_token",
+			expectedPath: "/realms/stacklok/protocol/openid-connect/revoke",
+			expectError:  false,
+			createServer: createTestServer,
```

Comment:
Another option here is to make createTestServer an anonymous function, which avoids needing to declare it or `TestCase` outside this function:

```suggestion
			createServer: func(t *testing.T) *httptest.Server {
				return httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
					require.NoError(t, r.ParseForm())
					require.Equal(t, "minder-cli", r.Form.Get("client_id"))
					require.Equal(t, "test-token", r.Form.Get("token"))
					require.Equal(t, "refresh_token", r.Form.Get("token_type_hint"))
				}))
			},
```